workbench-core/datasets/src/dataSetService.ts

```diff
@@ -50,16 +52,33 @@ export class DataSetService {
    * @returns the DataSet object which is stored in the backing datastore.
    */
   public async provisionDataSet(request: CreateProvisionDatasetRequest): Promise<DataSet> {
-    const { storageProvider, ...dataSet } = request;
-
-    await storageProvider.createStorage(dataSet.storageName, dataSet.path);
-
-    const provisioned: DataSet = {
-      ...dataSet,
-      storageType: storageProvider.getStorageType()
+    const metadata: Metadata = {
+      actor: { id: '-', roles: [] },
+      action: this.provisionDataSet.name,
+      source: {
+        serviceName: DataSetService.name
+      },
+      requestBody: request
     };
 
-    return await this._dbProvider.addDataSet(provisioned);
+    try {
+      const { storageProvider, ...dataSet } = request;
+
+      await storageProvider.createStorage(dataSet.storageName, dataSet.path);
+
+      const provisioned: DataSet = {
+        ...dataSet,
+        storageType: storageProvider.getStorageType()
+      };
+      const response: DataSet = await this._dbProvider.addDataSet(provisioned);
+      metadata.statusCode = 200;
```

Comment:
Should this be a 201 since we're creating a resource?

Reply:
I've asked that on other PRs. The response I've received is along the lines of "well, we don't determine a status code when we throw the error, and we leave it as an exercise to the consumer to choose a status code when they return a result. 200 and 400 are just supposed to represent generic success/failure."

I don't like it either. Working from the "we don't do status codes here" argument, I've removed the status code in the next push.